### ISSUE TITLE
[BUG] Trace viewer does not display tested page correctly

### ISSUE THREAD

**Issue by schotul** [microsoft/playwright#27839]
### System info
- Playwright Version: [v1.39.0]
- Operating System: [Windows 10]
- Browser: [Chromium]
- Other info:
  - Node js version v20.8.1
  - Confluence version 8.4.3

### Description
I am currently testing a page created with [Confluence](https://www.atlassian.com/software/confluence) (Data Center version). All the icons used by the Confluence UI are not displayed correctly in the Trace Viewer. Instead of the icon, a rectangle is displayed.

#### Example (detail and full screen)

| Expected result | Actual result |
| ---------- | --------- |
| ![confluence-screenshot-script-detail](https://github.com/microsoft/playwright/assets/149158978/50454742-f738-43cd-9c49-6fe77924a1c1) | ![confluence-screenshot-trace-viewer-detail](https://github.com/microsoft/playwright/assets/149158978/b5727ab4-32e9-498b-a770-71748a50df21) |
| <img src="https://github.com/microsoft/playwright/assets/149158978/5871928b-b86b-4b01-81f6-cb9c5ca05c4c" width="400px"> | <img src="https://github.com/microsoft/playwright/assets/149158978/c75f9205-fa8a-4af7-bcbb-c46ca7bdf117" width="400px"> |

### Source code
In a very simple example, the script opens a page on a Confluence test instance I am providing and then takes a screenshot.

- [x] I provided exact source code that allows reproducing the issue locally.

[files.zip](https://github.com/microsoft/playwright/files/13189180/files.zip)

**Config file**

```js
// playwright.config.ts
import { defineConfig, devices } from '@playwright/test';

export default defineConfig({
  testDir: 'test-confluence',
  projects: [
    {
      name: 'Google Chrome',
      use: {
        channel: 'chrome',
      },
    },
  ],
});
```

**Test file (self-contained)**

```js
import { test } from '@playwright/test';

test('confluence', async ({ page }) => {
  // Open Confluence test server instance.
  await page.goto('https://us01.swift-wiki.net/display/PT');

  // Create screenshot.
  await page.screenshot({ path: './confluence-screenshot-script.png' });
});
```

**Steps**
- [Run the test with _Show trace viewer_ activated in VS Code.]
- [Compare screenshot created by the script with display in Trace viewer.]

**Expected**
Trace viewer shows icons.

**Actual**
Trace viewer shows rectangles instead of the icons.

**Comment by yury-s** [comment]
Please share the trace file where it happens, so that we can investigate the issue.

**Comment by schotul** [comment]
Hi @yury-s , here it is: [trace.zip](https://github.com/microsoft/playwright/files/13192923/trace.zip)

**Comment by dgozman** [comment]
Investigation notes:
- When opening trace viewer, browser requests the `.ttf` version of the `@font-face`, while in the trace we have `.woff` version (that was probably requested by the browser while recording).
- So far, I don't see a way to query which particular url `@font-face` has resolved to, so cannot do a trick similar to `image.currentSrc`.

**Comment by schotul** [comment]
Hi @dgozman,

Since I wasn't sure if I understood you correctly, I investigated loading the font myself.

- When using the option _Show browser_ in VS Code, the font _adgs-icons.woff_ is loaded without problems and the icons are displayed correctly.
![font-loading-with-show-browser-option](https://github.com/microsoft/playwright/assets/149158978/0db48f37-a1d3-45a1-84ad-a2b011644aec)

- When using the option _Show trace viewer_ in VS Code (or by loading _trace.zip_ with [https://trace.playwright.dev/](https://trace.playwright.dev/)), the font _adgs-icons.woff_ is loaded without problems BUT after a while the browser also tries to load the font _adgs-icons.ttf_ and then fails.
![font-loading-with-show-trace viewer-option](https://github.com/microsoft/playwright/assets/149158978/e1bf1627-36ac-4f0a-8a81-66227ae04420)


In my opinion, answering the following questions would help to find a solution:

1. Why does the browser try to load _adgs-icons.ttf_ as well?
2. Why does this fail? After all, the URL of _adgs-icons.ttf_ is valid.

**Comment by schotul** [comment]
Hi @dgozman,

I've found the answer to question 1. from my previous post just by opening the console in Developer tools.

![image](https://github.com/microsoft/playwright/assets/149158978/15f07d7a-371e-4f81-a5c0-01deef966e0a)

- This shows that there is a problem loading the _adgs-icons.woff_ font. 
- Due to the @font-face rule in _batch.css_, the browser then tries to access the ttf font (but cannot find it).

The message _Failed to decode downloaded font_ could be caused by a corrupted font file. However, since the font file works if _Show browser_ is selected, the file on the server should be fine (not corrupted). Could the file have been corrupted during the loading process? Perhaps by using a transport method that is not suitable for binary files? Also, I still don't understand why accessing the ttf font results in a 404 error. Because manually copying the link into a browser tab downloads the ttf file without any problems.

Perhaps a crazy idea: **Is it possible that the snapshot was taken too early during recording? I mean _before the adgs-icons font was loaded correctly_?**

I recorded another trace and here I noticed something that was not visible in the first trace file. I used [trace.playwright.dev](https://trace.playwright.dev/), moved the mouse pointer slowly over the timeline to the right and checked the pop-up windows that appeared.

Here the results with [trace.zip](https://github.com/microsoft/playwright/files/13192923/trace.zip):

**Step 1:** At first, no icons are visible.

![image](https://github.com/microsoft/playwright/assets/149158978/36907cff-91f2-47c0-83af-de7885fb8932)

**Step 2:** Here the icons are correctly visible in the pop-up window but still displayed as rectangles in the screenshot in the background (please compare the icon right of _Spaces_). 

![image](https://github.com/microsoft/playwright/assets/149158978/cd2b2c6c-b8b0-40a9-b034-ebaffc95ed5e)

With the newly recorded trace file, I received an additional status in which the icons are also displayed as rectangles in the pop-up window.

[trace-new.zip](https://github.com/microsoft/playwright/files/13259447/trace-new.zip)

**Step 1:** At first, no icons are visible.

![image](https://github.com/microsoft/playwright/assets/149158978/52d9bc1a-8d0e-405e-8358-14f6f6eb5414)

**Step 2:** Icons shown as rectangles in popup window.

![image](https://github.com/microsoft/playwright/assets/149158978/172d2ec8-c984-4c82-8aa7-474bbb7ebd74)

**Step 3:** Here the icons are correctly visible in the pop-up window but still displayed as rectangles in the screenshot in the background.

![image](https://github.com/microsoft/playwright/assets/149158978/77bb7f5d-ff18-4b9b-b83e-4ce42e672177)

Since the screenshot in the background of the Trace Viewer always shows rectangles for the icons, I would assume that this screenshot was taken in step 2 and not in step 3.

Could this be possible?

**Comment by schotul** [comment]
Hi @dgozman,

The font file _adgs-icons.woff_, which is stored in _trace.zip_, is damaged and therefore cannot be loaded by the browser.

This can easily be seen by comparing the file size of the (only) file with extension _dat_ in _trace.zip_ and [adgs-icons.woff](https://us01.swift-wiki.net/s/-xkac1f/9011/1ljexkq/9.2.2/_/download/resources/com.atlassian.auiplugin:split_aui.splitchunk.56dfb54d0c/assets/adgs-icons.woff) that is available on the server.

![image](https://github.com/microsoft/playwright/assets/149158978/83c6b406-bb53-4a8c-b85d-6f60877f8313)

**Comment by dgozman** [comment]
@schotul You are right, it seems like this particular font is saved incorrectly.

**Comment by dgozman** [comment]
Upstream Chromium issue: https://bugs.chromium.org/p/chromium/issues/detail?id=1179646.

**Comment by schotul** [comment]
Hi @dgozman,

That's really interesting. I wouldn't have thought that the problem was related to the browser.

To doublecheck, I changed the **config file** to
```
// playwright.config.ts
import { defineConfig, devices } from '@playwright/test';

export default defineConfig({
	testDir: 'test-confluence',
	projects: [
		{
			name: 'Google Chrome',
			use: {
				channel: 'chrome',
			},
		},
		{
			name: 'firefox',
			use: {
				...devices['Desktop Firefox'],
			},
		},
		{
			name: 'webkit',
			use: {
				...devices['Desktop Safari'],
			},
		},
	],
});

```
and tested the Trace Viewer with Firefox and Webkit.

And as expected, the icons are correctly displayed with these browsers.
## Firefox
![grafik](https://github.com/microsoft/playwright/assets/149158978/c0c2bb27-6a3f-47db-b912-fd5609d2395e)

## Webkit
![grafik](https://github.com/microsoft/playwright/assets/149158978/b5c534a3-bfc7-4c6d-97ed-b446d1b82df6)

## Chrome
![grafik](https://github.com/microsoft/playwright/assets/149158978/2d8d0121-abe0-43c1-b3bd-afa381b64116)

**Comment by dgozman** [comment]
For the record, https://chromiumdash.appspot.com/commit/dd3de0fd74503f13b87bd238c405359c0b17413f should be released in Chromium 121, which should get into the next Playwright release.

**Comment by dgozman** [comment]
The fix will be released in v1.41.
